src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -490,16 +490,16 @@ XclBin::findAndReadMirrorData(std::fstream& _istream, boost::property_tree::ptre
   }
 
   // Bring the mirror metadata into memory
-  std::unique_ptr<unsigned char> memBuffer(new unsigned char[bufferSize]);
+  std::vector<unsigned char> memBuffer(bufferSize);
```

Comment:
I wonder whether we really need `unsigned char` and not just `char`.
This would prebent the use of all the casts to `char*` below.